components/sql/src/main/scala/pl/touk/nussknacker/sql/db/schema/MetaDataProviderFactory.scala

```diff
@@ -21,12 +20,15 @@ class MetaDataProviderFactory {
       props.put(k, v)
     }
     val ds = ThreadUtils.withContextClassLoader(getClass.getClassLoader) {
-      new DriverDataSource(
-        dbPoolConfig.url,
-        dbPoolConfig.driverClassName,
-        props,
-        dbPoolConfig.username,
-        dbPoolConfig.password
+      HikariDataSourceFactory(
+        dbPoolConfig.copy(
```

Comment:
We are using this config twice, so we should take care of generating a unique `poolName`, otherwise metrics will be broken

Reply:
I agree, these pool names were named `HikariPool-1` and `HikariPool-2` (depending on which was created first) - now they are explicitly named. 

I didn't add `poolName` to `DBPoolConfig` because it is a DTO for configuration. Instead, I used the enricher name + suffix. Do you think it is ok?

Why do you think that we should add `registerMbeans` to the configuration? You want to not expose mbeans for the pool created for metadata purposes? I wonder if it should be in the configuration or in `HikariDataSourceFactory.apply` parameters.

Reply:
I want to expose `registerMbeans` because it's disabled by default: https://github.com/brettwooldridge/HikariCP/wiki/MBean-(JMX)-Monitoring-and-Management (or at least it is according to docs)